src/mscorlib/corefx/Interop/Unix/System.Globalization.Native/Interop.Collation.cs

```diff
@@ -25,7 +25,7 @@ internal static partial class GlobalizationInterop
 
         [SecurityCritical]
         [DllImport(Libraries.GlobalizationInterop, CharSet = CharSet.Unicode, EntryPoint = "GlobalizationNative_IndexOf")]
-        internal unsafe static extern int IndexOf(SafeSortHandle sortHandle, string target, int cwTargetLength, char* pSource, int cwSourceLength, CompareOptions options);
+        internal unsafe static extern int IndexOf(SafeSortHandle sortHandle, string target, int cwTargetLength, char* pSource, int cwSourceLength, CompareOptions options, ref int matchLength);
```

Comment:
ok. my whole point is we need to be able to pass null (or IntPtr.Zero) from the managed side. I suggested IntPtr because we don't have to use unsafe code to do that. if we always pass non null we'll calculate the value which is not needed in most cases